funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -416,18 +416,53 @@ def stop_endpoint(
             exit(-1)
 
     @staticmethod
-    def delete_endpoint(endpoint_dir: pathlib.Path):
+    def delete_endpoint(
+        endpoint_dir: pathlib.Path, endpoint_config: Config | None, force: bool = False
+    ):
         ep_name = endpoint_dir.name
 
         if not endpoint_dir.exists():
             log.warning(f"Endpoint <{ep_name}> does not exist")
             exit(-1)
 
-        # stopping the endpoint should handle all of the process cleanup before
-        # deletion of the directory
+        endpoint_id = Endpoint.get_endpoint_id(endpoint_dir)
+        if endpoint_id is None:
+            log.warning(f"Endpoint <{ep_name}> could not be located")
+            if not force:
+                exit(-1)
+
+        # Delete endpoint from web service
+        try:
+            fx_client = Endpoint.get_funcx_client(endpoint_config)
+            fx_client.delete_endpoint(endpoint_id)
+        except GlobusAPIError as e:
+            log.warning(
+                f"Endpoint <{ep_name}> could not be deleted from web service."
+                + f"  [{e.message}]"
+            )
+            if not force:
```

Comment:
Thinking about whether we should explain more explicitly that we exited and didn't stop or remove anything in these two cases? 
ie. "Not deleting <ep_name>, .....<reason>"

Reply:
That's a good point. I've added an additional critical log message that should bring some more clarity,